### PR TITLE
Added notes of existing results, upgrade version of JMH, remove properties which can be defaulted, change warm up iterations.

### PR DESCRIPTION
**Pull request by nikhilnanivadekar**

### CODE REVIEW

**Comment by nikhilnanivadekar** [comment]
FYI with the upgraded version of JMH, JMH will capture and print VM details: https://hg.openjdk.java.net/code-tools/jmh/rev/9df26c5f8b4b

**Comment by nikhilnanivadekar** [comment]
Refer to this changelog for the current defaults: https://hg.openjdk.java.net/code-tools/jmh/rev/56576f4dc727
It is advisable to keep the timeout and measurement time to be at default unless these are long running iterations.
Secondly, it is considered good practice to keep the warm-up iteration twice as measurement iterations. This will reduce the additional noise and deviations observed.

File: README.md [hunk 1]
```diff
@@ -123,4 +123,13 @@ From the root folder, run:
...
+``` 
+
+### Experimenting with Results
+If you wish to experiment with the results of the benchmarks we have already run you can look at the folders:
+`results_for_1k_size` and `results_for_1M_size`.
+Notes about the results:
+* The results for each benchmark run is saved in it's own respective folder.
+* The results are saved as json files as well as maven build output.
+* For json files navigate to benchmark-results (For Eg. /results_for1k_size/benchmark-results/int-lists-filter)
+* For maven build output navigate to output (For Eg. /results_for_1k_size/output/)
```
Comment (c-guntur) [inline README.md]: NIce !

File: README.md [hunk 2]
```diff
@@ -123,4 +123,13 @@ From the root folder, run:
...
\ No newline at end of file
+``` 
+
+### Experimenting with Results
+If you wish to experiment with the results of the benchmarks we have already run you can look at the folders:
+`results_for_1k_size` and `results_for_1M_size`.
+Notes about the results:
+* The results for each benchmark run is saved in it's own respective folder.
+* The results are saved as json files as well as maven build output.
+* For json files navigate to benchmark-results (For Eg. /results_for1k_size/benchmark-results/int-lists-filter)
```
Comment (c-guntur) [inline README.md]: missing one underscore in the folder name between for and 1k. Please replace: results_for1k_size with results_for_1k_size

**Comment by nikhilnanivadekar** [comment]
fixed and rebased.

**Comment by c-guntur** [comment]
Can you please separate these as different PRs?

**Comment by nikhilnanivadekar** [comment]
I can but would like to know why?
These are separate commits, they show up separately on the network graph. Separating PRs would mean after every merge I need to rebase. Don’t quite follow the workflow here.

**Pull request merged**